lib/logstash/outputs/sqs.rb

```diff
@@ -1,136 +1,174 @@
-#              "sqs:ChangeMessageVisibility",
-#              "sqs:ChangeMessageVisibilityBatch",
-#              "sqs:GetQueueAttributes",
-#              "sqs:GetQueueUrl",
-#              "sqs:ListQueues",
-#              "sqs:SendMessage",
-#              "sqs:SendMessageBatch"
-#            ],
-#            "Effect": "Allow",
-#            "Resource": [
-#              "arn:aws:sqs:us-east-1:200850199751:Logstash"
-#            ]
-#          }
-#        ]
-#      }
+#   * The maximum allowed individual message size is 256KiB.
+#   * The maximum total payload size (i.e. the sum of the sizes of all
+#     individual messages within a batch) is also 256KiB.
 #
-# See http://aws.amazon.com/iam/ for more details on setting up AWS identities.
+# This plugin will dynamically adjust the size of the batch published to SQS in
+# order to ensure that the total payload size does not exceed 256KiB.
+#
+# WARNING: This output cannot currently handle messages larger than 256KiB. Any
+# single message exceeding this size will be dropped.
 #
 class LogStash::Outputs::SQS < LogStash::Outputs::Base
-  include LogStash::PluginMixins::AwsConfig
-  include Stud::Buffer
+  include LogStash::PluginMixins::AwsConfig::V2
 
-  config_name "sqs"
+  config_name 'sqs'
+  default :codec, 'json'
 
-  # Name of SQS queue to push messages into. Note that this is just the name of the queue, not the URL or ARN.
-  config :queue, :validate => :string, :required => true
+  concurrency :shared
 
-  # Set to true if you want send messages to SQS in batches with `batch_send`
-  # from the amazon sdk
-  config :batch, :validate => :boolean, :default => true
+  # Set to `true` to send messages to SQS in batches (with the
+  # `SendMessageBatch` API) or `false` to send messages to SQS individually
+  # (with the `SendMessage` API). The size of the batch is configurable via
+  # `batch_events`.
+  config :batch, :validate => :boolean, :default => true, :deprecated => true
 
-  # If `batch` is set to true, the number of events we queue up for a `batch_send`.
+  # The number of events to be sent in each batch. This is only relevant when
+  # `batch` is set to `true`.
   config :batch_events, :validate => :number, :default => 10
 
-  # If `batch` is set to true, the maximum amount of time between `batch_send` commands when there are pending events to flush.
-  config :batch_timeout, :validate => :number, :default => 5
+  config :batch_timeout, :validate => :number, :deprecated => 'This setting no longer has any effect.'
 
-  public
-  def aws_service_endpoint(region)
-    return {
-        :sqs_endpoint => "sqs.#{region}.amazonaws.com"
-    }
-  end
+  # The maximum number of bytes for any message sent to SQS. Messages exceeding
+  # this size will be dropped. See
+  # http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/limits-messages.html.
+  config :message_max_size, :validate => :bytes, :default => '256KiB'
 
-  public 
+  # The name of the target SQS queue.
```

Comment:
maybe add back `Note that this is just the name of the queue, not the URL or ARN.` ?